haystack/nodes/file_converter/pdf.py

```diff
@@ -252,7 +251,7 @@ def convert(
         try:
             images = convert_from_path(file_path)
             for image in images:
-                temp_img = tempfile.NamedTemporaryFile(dir=os.path.dirname(os.path.realpath(__file__)), suffix=".jpeg")
+                temp_img = tempfile.NamedTemporaryFile(suffix=".jpeg")
```

Comment:
@ZanSara aren't you going to get the same files in every loop (because you use suffixes)? I am assuming that `convert_from_path` generates multiple image files.

Reply:
`tempfile.NamedTemporaryFile(suffix=".jpeg")` generates a temporary file for every iteration, so I don't see the file being reused in any way. 

But to be honest, even if the filename was always the same (which I doubt), why would that pose an issue? The image is converted on the fly, the next iteration won't need the previous iteration's file, nor it will be any useful later.

Reply:
```python
from tempfile import NamedTemporaryFile

for i in range(10):
    print(NamedTemporaryFile(suffix=".jpeg").name)
```
```
/tmp/tmp0usx47fg.jpeg
/tmp/tmpe8k4b2n5.jpeg
/tmp/tmp8dpmo2yy.jpeg
/tmp/tmp4zdr5fqc.jpeg
/tmp/tmpyfo9z285.jpeg
/tmp/tmpbjgzovts.jpeg
/tmp/tmpz_osgmhx.jpeg
/tmp/tmpq594uvqc.jpeg
/tmp/tmp8dys1pmd.jpeg
/tmp/tmpwr_d4ym2.jpeg
```

Reply:
Right, I get what's going on now.